File: a1-service/src/main/java/a1/a1service/dto/ObrazacAutorskoDeloCreationDTO.java
package a1.a1service.dto;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
public class ObrazacAutorskoDeloCreationDTO {

    @XmlElement
    private FizickoLiceDTO fizickoLiceDTO;

    @XmlElement
    private PravnoLiceDTO pravnoLiceDTO;

    @XmlElement
    private PunomocnikDTO punomocnikDTO;

    @XmlElement
    private PodaciOAutorskomDeluDTO podaciOAutorskomDeluDTO;

    @XmlElement
    private PriloziUzPrijavuDTO priloziUzPrijavuDTO;

    public ObrazacAutorskoDeloCreationDTO() {
    }

    public FizickoLiceDTO getFizickoLiceDTO() {
        return fizickoLiceDTO;
    }

    public PravnoLiceDTO getPravnoLiceDTO() {
        return pravnoLiceDTO;
    }

    public PunomocnikDTO getPunomocnikDTO() {
        return punomocnikDTO;
    }

    public PodaciOAutorskomDeluDTO getPodaciOAutorskomDeluDTO() {
        return podaciOAutorskomDeluDTO;
    }

    public PriloziUzPrijavuDTO getPriloziUzPrijavuDTO() {
        return priloziUzPrijavuDTO;
    }
}

File: xml-front/src/app/models/autorska/ObrazacAutorskoDeloCreationDTO.ts
import { FizickoLiceDTO } from "./FizickoLiceDTO";
import { PodaciOAutorskomDeluDTO } from "./PodaciOAutorskomDeluDTO";
import { PravnoLiceDTO } from "./PravnoLiceDTO";
import { PriloziUzPrijavuDTO } from "./PriloziUzPrijavuDTO";
import { PunomocnikDTO } from "./PunomocnikDTO";

export interface ObrazacAutorskoDeloCreationDTO{
    fizickoLiceDTO?: FizickoLiceDTO;
    pravnoLiceDTO?: PravnoLiceDTO;
    punomocnikDTO?: PunomocnikDTO;
    podaciOAutorskomDeluDTO: PodaciOAutorskomDeluDTO;
    priloziUzPrijavuDTO?: PriloziUzPrijavuDTO;
    podnosilacAutor: boolean;
}